src/lib/userStorage/UserProperties.js

```diff
@@ -61,17 +63,16 @@ export default class UserProperties {
       this.local = assign({}, localProps)
       this.throttlePersist = throttle(() => this.persist(), 5000, { leading: true, trailing: true })
 
-      log.debug('found local settings:', { props, localProps, local: this.local })
-
-      // if not props then block
-      if (isNil(props)) {
-        await this._syncFromRemote()
-      } else {
-        // otherwise sync withs storage in background
-        this._syncProps(props)
-        this._syncFromRemote()
+      // Sync with local properties in this.ready only if they are set during app startup/sign-in,
+      // and remote properties synchronization is still in progress.
+      // In such cases, wait for remote sync to complete before syncing local properties.
+      // Otherwise, we only want to syncFromRemote during initialization.
+      if (mutex.isLocked()) {
+        await withMutex(mutex, () => this._syncProps(props))
+        return this.data
       }
 
+      await withMutex(mutex, () => this._syncFromRemote())
```

Comment:
this is incorrect.
1. there's no need to check mutex.isLocked
2. you didnt fix anything, remote will still override local props

correct mutex usage
when reading/writing from storage you want to make sure actions do not conflict and finish before another read/write is called.
1. the method syncFromRemote and Persist should both acquire the lock as first step
2. in order to make sure any `set` is not overwritten by syncFromRemote, the set method should also acquire the lock before updating the props
3. finally there's no reason to call both syncProps and syncFromRemote in the constructor ready because you `await` so syncFromRemote will always be the final result in props. syncProps have no meaning.

If the above solution works for you it means the only thing you fixed is that you always await on syncFromRemote in constructor.
I think the simple solution is simply to always await on syncFromRemote in ready. there's no need for mutex.